lifecycle/lifecycle/event_stream/server.py

```diff
@@ -0,0 +1,71 @@
+import asyncio
+import threading
+
+import socketio
+import time
+
+from lifecycle.config import Config
+from lifecycle.job.registry import list_job_registry
+from racetrack_client.log.logs import get_logger
+from racetrack_commons.entities.dto import JobDto
+
+logger = get_logger(__name__)
+
+
+class EventStreamServer:
+    def __init__(self, config: Config):
+        """Socket.IO server for streaming events to clients"""
+        self.sio = socketio.AsyncServer(async_mode='asgi', cors_allowed_origins='*')
+        self.clients: list[str] = []  # List of Client IDs
+        self.watcher_thread: threading.Thread | None = None
+        self.config = config
+
+        @self.sio.event
+        async def connect(client_id: str, environ):
+            logger.debug(f'Client {client_id} connected to Event Streamer')
+            self.clients.append(client_id)
+            await self.sio.emit('connected', {}, to=client_id)
+            if len(self.clients) > 0 and (self.watcher_thread is None or not self.watcher_thread.is_alive()):
+                self.watcher_thread = threading.Thread(target=self.watch_database_events, args=(), daemon=True)
+                self.watcher_thread.start()
+
+        @self.sio.event
+        async def disconnect(client_id: str):
+            logger.debug(f'Client {client_id} disconnected from Event Streamer')
+            if client_id in self.clients:
+                self.clients.remove(client_id)
+
+        @self.sio.on('*')
+        async def catch_all(event, sid, data):
+            logger.warning(f'Unhandled Socket.IO event: {event}, {sid}, {data}')
+
+        self.asgi_app = socketio.ASGIApp(self.sio, socketio_path='lifecycle/socketio/events')
```

Comment:
The socketio_path could be init parameter, as I see you're trying to have this class be generic (except for watch_database_events, which would require passing callbacks, but maybe not worth this time).